src/rgw/rgw_sal_motr.cc

```diff
@@ -2178,65 +2235,84 @@ int MotrObject::get_bucket_dir_ent(const DoutPrefixProvider *dpp, rgw_bucket_dir
                           << rc << dendl;
         return rc;
       }
-
       iter = bl.cbegin();
       ent.decode(iter);
       rc = 0;
       // Put into the cache
       this->store->get_obj_meta_cache()->put(dpp, this->get_name(), bl);
       goto out;
-
     }
     else
-    {  // Version-id instance is empty
-       // Cache miss.
-        keys[0] = this->get_name();
-
-        // Retrieve all 'max' number of pairs.
-        rc = store->next_query_by_name(bucket_index_iname, keys, vals, this->get_name());
-        if (rc < 0) {
-          ldpp_dout(dpp, 0) << __func__ << "ERROR: NEXT query failed. " << rc << dendl;
-          return rc;
-        }
-
-        rc = -ENOENT;
-
-        // Iterating on object keys and return the latest object version
-        for (const auto& bl: vals) {
-          if (bl.length() == 0)
-            break;
+    {
+      // Version-id instance is empty
+      // Cache miss.
+      keys[0] = this->get_name();
+      // Retrieve all 'max' number of pairs.
+      rc = store->next_query_by_name(bucket_index_iname, keys, vals, this->get_name());
+      if (rc < 0) {
+        ldpp_dout(dpp, 0) << __func__ << "ERROR: NEXT query failed. " << rc << dendl;
+        return rc;
+      }
 
-          iter = bl.cbegin();
-          ent.decode(iter);
+      // Iterating on object keys and return the latest object version
+      int i = 1;
+      for (; i < rc; ++i) {
+        if (vals[i].length() == 0) 
```

Comment:
> if (vals[i].length() == 0) 

why latest version has this condition?

Reply:
It is not required now. removed this code.